pylint/checkers/base/basic_checker.py

```diff
@@ -658,12 +658,26 @@ def _check_misplaced_format_function(self, call_node: nodes.Call) -> None:
                 self.add_message("misplaced-format-function", node=call_node)
 
     @utils.only_required_for_messages(
-        "eval-used", "exec-used", "bad-reversed-sequence", "misplaced-format-function"
+        "eval-used",
+        "exec-used",
+        "bad-reversed-sequence",
+        "misplaced-format-function",
+        "unreachable",
     )
     def visit_call(self, node: nodes.Call) -> None:
-        """Visit a Call node -> check if this is not a disallowed builtin
-        call and check for * or ** use.
+        """Visit a Call node ->
+        1
+
+        check if this is not a disallowed builtin call
+        2. check for * or ** use.
+        3. check if this call is sys.exit and is followed by unreachable code
         """
+        if (
+            isinstance(node.func, nodes.Attribute)
+            and node.func.attrname == "exit"
```

Comment:
Won't this catch `other_module.exit()`?

Reply:
Yes, we need to use inference for this kind of thing.

Reply:
Exactly the feedback I was looking for. Pierre, can you explain what you mean by use inference? Do you mean the INFERENCE confidence level?

Reply:
Well yes, but when we use the inference confidence level it means we used inference to come to a conclusion. Here's an example of inference (notice the ``infer()`` call) :
```python
>>> import astroid
>>> a = astroid.extract_node('''
... import argparse
... 
... parser = argparse.ArgumentParser()
... parser.add_argument(
...     "config_folder", help="Path to the project configuration folder"
... )
... args = parser.parse_args()
... try:
...     int(args.config_folder) / 0
... except:
...     parser.error(
...         f"Your configuration folder can't be divided by zero, use something else !"
...     )
... ''')
>>> list(a.handlers[0].body[0].value.func.infer())[0]
<BoundMethod error of argparse.ArgumentParser at 0x140506382647696>
```

Reply:
AHA! You just answered one of my top questions about pylint / astroid. I couldn't figure out how the code would distinguish between a call to `method_a` if this method name could be defined in `module1.method_a` and `module2.method_a` looks like a call to `infer` is the way!